python/packages/nisar/products/insar/utils.py

```diff
@@ -370,6 +372,128 @@ def _compute_subswath_mask_id(azi_idx,
+    ---------
+    radar_grid_obj : SLC
+        The radar grid object for the reference RSLC
+    orbit_obj : isce3.core.Orbit
+        The SLC object for the secondary RSLC
+    dem_file  : str
+        DEM file
+    out_dem_rdr_path : str
+        output path of the DEM in radar grid
+    use_gpu : boolean (default: True)
+        Indicator to use the GPU or not
+    dem_interp_method : str (default: BIQUINTIC)
+        DEM interpolation method, one of 'BILINEAR', 'BICUBIC', 'NEAREST', and 'BIQUINTIC'
+    threshold : float (default: 1.0e-7)
+        The rdr2geo absolute slant range convergence tolerance (m)
+    numiter : integer (default: 25)
+        Maximum number of primary Newton-Raphson iterations
+    extraiter : integer (default: 10)
+         Maximum number of secondary iterations
+    lines_per_block : integer (default: 1000)
+         Lines per block to write the generated DEM to the hard drive
+    """
+
+    error_journal = journal.error('utils.generate_insar_dem')
+    grid_doppler = isce3.core.LUT2d()
+
+    dem_raster = isce3.io.Raster(dem_file)
+    if dem_raster is None:
+        err_str = f'Can not open the DEM file {dem_raster}'
+        error_journal.log(err_str)
+        raise ValueError(err_str)
+    epsg = dem_raster.get_epsg()
+    proj = isce3.core.make_projection(epsg)
+    ellipsoid = proj.ellipsoid
+
+    # Default DEM interpolation method is BIQUINTIC
+    interp_method = isce3.core.DataInterpMethod.BIQUINTIC
+    if dem_interp_method == 'BILINEAR':
+        interp_method = isce3.core.DataInterpMethod.BILINEAR
+    if dem_interp_method == 'BICUBIC':
+        interp_method = isce3.core.DataInterpMethod.BICUBIC
+    if dem_interp_method == 'NEAREST':
+        interp_method = isce3.core.DataInterpMethod.NEAREST
+
+    # Use the GPU or CPU version
+    if use_gpu:
+        Rdr2Geo = isce3.cuda.geometry.Rdr2Geo
+    else:
+        Rdr2Geo = isce3.geometry.Rdr2Geo
+
+    # Create the DEM in the range Doppler coordinates
+    dem_src = isce3.io.Raster(out_dem_rdr_path,
+                              radar_grid_obj.width,
+                              radar_grid_obj.length, 1,
+                              gdal.GDT_Float32, 'ENVI')
+
+    # Build the Rdr2Geo object
+    rdr2geo_obj = Rdr2Geo(radar_grid_obj, orbit_obj, ellipsoid, grid_doppler,
+                          dem_interp_method=interp_method,
+                          threshold=threshold, numiter=numiter,
+                          extraiter=extraiter,
+                          lines_per_block=lines_per_block)
+
+    x_raster, y_raster, incidence_raster,\
+        heading_raster, local_incidence_raster, local_psi_raster,\
+            simulated_amplitude_raster, shadow_raster = [None] * 8
+    rdr2geo_obj.topo(dem_raster, x_raster, y_raster, dem_src,
+                     incidence_raster, heading_raster, local_incidence_raster,
+                     local_psi_raster, simulated_amplitude_raster,
+                     shadow_raster, None, None)
```

Comment:
Not a big deal but for clarity which rasters are you ignoring. 
```suggestion
            simulated_amplitude_raster, shadow_raster, ground_to_sat_x_ratser, ground_to_sat_y_raster = [None] * 10
    rdr2geo_obj.topo(dem_raster, x_raster, y_raster, dem_src,
                     incidence_raster, heading_raster, local_incidence_raster,
                     local_psi_raster, simulated_amplitude_raster,
                     shadow_raster, ground_to_sat_x_ratser, ground_to_sat_y_raster )
```